### PR TITLE
Code review prompt update

### PR DESCRIPTION
**Pull request by olewicki**
- transforming references from Firefox to generalized term
- filtering on praising/descriptive comments

### CODE REVIEW

File: bugbug/tools/code_review.py [hunk 1]
```diff
@@ -56,25 +56,37 @@ class HunkNotInPatchError(ModelResultError):
     """Occurs when the hunk in the model result is not part of the patch."""
 
 
-PROMPT_TEMPLATE_SUMMARIZATION = """You are an expert reviewer for the Mozilla Firefox source code, with experience on source code reviews.
+COND_MENTION_MOZILLA = False
```
Comment (marco-c) [inline bugbug/tools/code_review.py:59]:
```suggestion
TARGET_SOFTWARE= None
```
and then replace `(" the Mozilla Firefox" if COND_MENTION_MOZILLA else "")` with `(f" the {TARGET_SOFTWARE}" if TARGET_SOFTWARE is not None else "")`

This way it's more generic.

File: bugbug/tools/code_review.py [hunk 2]
```diff
@@ -86,21 +98,28 @@ class HunkNotInPatchError(ModelResultError):
...
 - applicable for the patch;
-- consistent with the source code in Firefox;
-- focusing on reporting possible bugs, major readability regressions, or similar concerns.
+- consistent with the source code"""
+    + (" in Firefox" if COND_MENTION_MOZILLA else "")
+    + """;
+- focusing on reporting possible bugs, major readability regressions, or similar concerns;
+- filter out any descriptive comments;
+- filter out any praising comments;
+- filter out any comments that do not suggest code changes.
```
Comment (marco-c) [inline bugbug/tools/code_review.py:111]: This last one might be too strict, have you tested it?
We might want to test it on our data to make sure it is not filtering out too much.

Comment (olewicki) [reply bugbug/tools/code_review.py:111]: Actually on the few examples I tested it's mostly lines 109-110 and the example line 137 that had an impact, we could remove line 111

Comment (marco-c) [reply bugbug/tools/code_review.py:111]: Let's remove it for now then

File: bugbug/tools/code_review.py [hunk 3]
```diff
@@ -968,13 +988,23 @@ def run(self, patch: Patch) -> list[InlineComment] | None:
...
+                    else " in the development community"
+                )
+                + "."
             },
-            {"output": "Sure, I'm aware of source code practices in Firefox."},
         )
         memory.save_context(
             {
-                "input": 'Please, analyze the code provided and report a summarization about the new changes; for that, focus on the code added represented by lines that start with "+". '
+                "input": 'Please, analyze the code provided and report a summarization about the new changes; for that, focus on the code added represented by lines that start with "+". \n'
```
Comment (marco-c) [inline bugbug/tools/code_review.py:1007]: nit: remove whitespace ```suggestion
                "input": 'Please, analyze the code provided and report a summarization about the new changes; for that, focus on the code added represented by lines that start with "+".\n'
```

Comment (olewicki) [reply bugbug/tools/code_review.py]: done

File: bugbug/tools/code_review.py [hunk 4]
```diff
@@ -56,25 +56,37 @@ class HunkNotInPatchError(ModelResultError):
...
+    """You will be given a task to generate a code review for the patch below. Use the following steps to solve it:
 1. Understand the changes done in the patch by reasoning about the summarization as previously reported.
 2. Identify possible code snippets that might result in possible bugs, major readability regressions, and similar concerns.
-3. Reason about each identified problem to make sure they are valid. Have in mind, your review must be consistent with the source code in Firefox.
+3. Reason about each identified problem to make sure they are valid. Have in mind, your review must be consistent with the source code"""
+    + (f" in {TARGET_SOFTWARE}" if TARGET_SOFTWARE is not None else "")
+    + """.
 4. Filter out comments that focuses on documentation, comments, error handling, tests, and confirmation whether objects, methods and files exist or not.
-5. Final answer: Write down the comments and report them using the JSON format provided below as valid comment examples (return only a JSON array).
+5. Filter out comments that are descriptive and filter out comments that are praising (example: "This is a good addition to the code.").
```
Comment (marco-c) [inline bugbug/tools/code_review.py:79]: Just realized now: should we split this into two (one about descriptive and one about praising), or keep only one?

Comment (olewicki) [reply bugbug/tools/code_review.py:79]: I can split it to make sure the model can focus on both

Comment (olewicki) [reply bugbug/tools/code_review.py]: done

**Review by marco-c — APPROVED** [review]

**Pull request merged**